src/generic/PermGroups.jl

```diff
@@ -642,36 +644,31 @@ end
 
 Base.eltype(::Type{SymmetricGroup{T}}) where T = Perm{T}
 
-Base.length(G::SymmetricGroup) = order(G)
+Base.length(G::SymmetricGroup) = order(Int, G)
 
 ###############################################################################
 #
 #   Misc
 #
 ###############################################################################
 
-@doc Markdown.doc"""
-    order(G::SymmetricGroup) -> BigInt
-
-Return the order of the full permutation group as `BigInt`.
-"""
-order(G::SymmetricGroup) = order(BigInt, G)
-order(::Type{T}, G::SymmetricGroup) where T = factorial(T(G.n))
-
-@doc Markdown.doc"""
-    order(a::Perm) -> BigInt
+function gens(G::SymmetricGroup{I}) where {I}
+   G.n == 1 && return eltype(G)[]
+   if G.n == 2
+      a = one(G)
+      a[1], a[2] = 2, 1
+      return [a]
+   end
+   a, b = one(G), one(G)
+   circshift!(a.d, b.d, -1)
+   b[1], b[2] = 2, 1
+   return [a, b]
+end
 
-Return the order of permutation `a` as `BigInt`.
+order(::Type{I}, G::SymmetricGroup) where {I} = convert(I, factorial(G.n))
```

Comment:
Hmm, perhaps like this? Otherwise you'll get overflows if `G.n` is big, no?
```suggestion
order(::Type{T}, G::SymmetricGroup) where {I} = convert(T, factorial(T(G.n)))
```